### ISSUE TITLE
Could it be possible to use a more restrictive scope for Google Drive access?

### ISSUE THREAD

**Issue by mably** [Kyrodan/KeeAnywhere#109]
I was wondering why we are using the "https://www.googleapis.com/auth/drive" OAuth scope which gives access to all the user's files and folders instead of the more restrictive one "https://www.googleapis.com/auth/drive.file" who restricts access to the files which have been created by the app.

https://github.com/Kyrodan/KeeAnywhere/blob/e53a28bb70815fe38226fba291439dce92c29834/KeeAnywhere/StorageProviders/GoogleDrive/GoogleDriveHelper.cs#L35

Source: https://developers.google.com/drive/v2/web/about-auth

What do you think?  Would definitely feel more secure, and would work at least when a new file is being created.

**Comment by Kyrodan** [comment]
Hmm, it's the same discussion like for Dropbox, see issue #24 
I will take a look wether this is possible for Google Drive, too.

**Comment by mably** [comment]
Here is message displayed by Google when such permission is required:

https://i.imgur.com/38OXySe.png

Looks like a perfect fit to me.

**Comment by eulercamposbarros** [comment]
I tried all suitable scopes and none allowed all the functionalities to keep running. Here's an overview of the issues I found:

- The _drive.file_ scope only allow KeeAnywhere to access files that it created but for major cases, the user uploads the database to Drive and want to access it using KeeAnywhere.
- The _drive.readonly_ scope allows it to read the database but it won't be able to sync new changes on the database.
